providers/default.rb

```diff
@@ -76,12 +76,16 @@ def load_prerequisites!
   load_mixlib_versioning
 end
 
+def client_flavor(chef_root)
+  chef_root =~ /angrychef/ ? 'angrychef' : 'chef'
+end
+
 def mixlib_install
   load_mixlib_install
   detected_platform = Mixlib::Install.detect_platform
   Chef::Log.debug("Platform detected as #{detected_platform} by mixlib_install")
   options = {
-    product_name: 'chef',
+    product_name: client_flavor(node['chef_packages']['chef']['chef_root']),
```

Comment:
coupling the product name to the root is very likely another case of tight coupling.  should be two attributes, and one should be the full path "/opt/angrychef" and the other should be the product_name "angrychef".